src/System.Management.Automation/engine/InvocationInfo.cs

```diff
@@ -294,7 +294,7 @@ public string PSScriptRoot
                 }
                 else
                 {
-                    return string.Empty;
+                    return MyCommand.Context.SessionState.Path.CurrentFileSystemLocation.ProviderPath;
```

Comment:
Conceptually this is incorrect because there can be a fileless pwsh.
If we want to create a truly modular pwsh, the file provider should not be mandatory.

Reply:
I've reverted this change. I didn't realize there was a difference between the `PSScriptRoot` variable and property. Here's the relevant text from the automatic variables help topic:
> Unlike the $PSScriptRoot and $PSCommandPath automatic variables, the PSScriptRoot and PSCommandPath properties of the $MyInvocation automatic variable contain information about the invoker or calling script, not the current script.